src/components/Settings/Language/Language.container.js

```diff
@@ -91,6 +109,249 @@ export class LanguageContainer extends Component {
+              .length > 1;
+
+          const langFullCode = showLangCode ? `(${lang})` : '';
+          langObject.nativeName = `${langObject.nativeName} ${langFullCode}`;
+          return langObject;
+        });
+      };
+      const downloadablesLangsArray = downloadablesTts.map(tts => {
+        return { langs: tts.langs, marketId: tts.marketId, ttsName: tts.name };
+      });
+      const identifiedLangsArray = downloadablesLangsArray.map(langObject =>
+        langObject.langs.map(language => {
+          return {
+            lang: language,
+            marketId: langObject.marketId,
+            ttsName: langObject.ttsName
+          };
+        })
+      );
+      const INITIAL_VALUE = [];
+      const downloadablesLangs = identifiedLangsArray.reduce(
+        (accumulator, currentValue) => accumulator.concat(currentValue),
+        INITIAL_VALUE
+      );
+      return formatLangObject(downloadablesLangs);
+    };
+
+    const filterAvailablesAndDownloadablesLangs = downloadablesLangs => {
+      const { langs } = this.props;
+      const slicedLangs = langs.map(lang => lang.slice(0, 2));
+      return downloadablesLangs.filter(({ langCode }) =>
+        slicedLangs.includes(langCode)
+      );
+    };
+
+    const filterDownloadablesOnlyLangs = (
+      downloadables,
+      availableAndDownloadables
+    ) => {
+      return downloadables.filter(
+        downloadableLang =>
+          !availableAndDownloadables.includes(downloadableLang)
+      );
+    };
+
+    const downloadablesLangsList = getDownloadablesLenguages(downloadablesTts);
+    const avaliableAndDownloadablesLangs = filterAvailablesAndDownloadablesLangs(
+      downloadablesLangsList
+    );
+    const downloadablesOnly = filterDownloadablesOnlyLangs(
+      downloadablesLangsList,
+      avaliableAndDownloadablesLangs
+    );
+    return {
+      avaliableAndDownloadablesLangs,
+      downloadablesOnly
+    };
+  };
+
+  pauseCallback = () => {
+    navigator.app.exitApp();
+  };
+
+  langOnAvailableTtsClick = async (event, downloadingLangData) => {
+    const { setDownloadingLang, ttsEngine } = this.props;
+    const { ttsName, lang, marketId } = downloadingLangData;
+    if (ttsEngine.name === ttsName) {
+      const downloadingLangState = {
+        isdownloading: true,
+        isDiferentTts: false,
+        engineName: ttsName,
+        marketId: marketId,
+        selectedLang: lang
+      };
+      setDownloadingLang(downloadingLangState);
+      this.setState({
+        downloadingLangError: {
+          ttsError: false,
+          langError: true
+        }
+      });
+      return;
+    }
+    const downloadingLangState = {
+      isdownloading: true,
+      isDiferentTts: true,
+      engineName: ttsName,
+      marketId: marketId,
+      selectedLang: lang
+    };
+    setDownloadingLang(downloadingLangState);
+    await this.handleSetTtsEngine(ttsName); //after tts change it fires a remounting
```

Comment:
this does not happen if I select a language from the downloadable list and have no voice available in the corresponding tts.

Reply:
good eye. thanks :)